pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/InTransformFunction.java

```diff
@@ -0,0 +1,381 @@
+    for (int i = 1; i < numArguments; i++) {
+      TransformFunction valueFunction = arguments.get(i);
+      if (valueFunction instanceof LiteralTransformFunction) {
+        stringValues.add(((LiteralTransformFunction) valueFunction).getLiteral());
+      } else {
+        allLiteralValues = false;
+        break;
+      }
+    }
+
+    if (allLiteralValues) {
+      int numValues = stringValues.size();
+      DataType storedType = _mainFunction.getResultMetadata().getDataType().getStoredType();
+      switch (storedType) {
+        case INT:
+          IntOpenHashSet intValues = new IntOpenHashSet(numValues);
+          for (String stringValue : stringValues) {
+            intValues.add(Integer.parseInt(stringValue));
+          }
+          _valueSet = intValues;
+          break;
+        case LONG:
+          LongOpenHashSet longValues = new LongOpenHashSet(numValues);
+          for (String stringValue : stringValues) {
+            longValues.add(Long.parseLong(stringValue));
+          }
+          _valueSet = longValues;
+          break;
+        case FLOAT:
+          FloatOpenHashSet floatValues = new FloatOpenHashSet(numValues);
+          for (String stringValue : stringValues) {
+            floatValues.add(Float.parseFloat(stringValue));
+          }
+          _valueSet = floatValues;
+          break;
+        case DOUBLE:
+          DoubleOpenHashSet doubleValues = new DoubleOpenHashSet(numValues);
+          for (String stringValue : stringValues) {
+            doubleValues.add(Double.parseDouble(stringValue));
+          }
+          _valueSet = doubleValues;
+          break;
+        case STRING:
+          _valueSet = stringValues;
+          break;
+        case BYTES:
+          ObjectOpenHashSet<ByteArray> bytesValues = new ObjectOpenHashSet<>(numValues);
+          for (String stringValue : stringValues) {
+            bytesValues.add(BytesUtils.toByteArray(stringValue));
+          }
+          _valueSet = bytesValues;
+          break;
+        default:
+          throw new IllegalStateException();
+      }
+    } else {
+      Preconditions.checkArgument(_mainFunction.getResultMetadata().isSingleValue(),
+          "The first argument for IN transform function must be single-valued when there are non-literal values");
+      _valueFunctions = new TransformFunction[numArguments - 1];
+      for (int i = 1; i < numArguments; i++) {
+        TransformFunction valueFunction = arguments.get(i);
+        Preconditions.checkArgument(valueFunction.getResultMetadata().isSingleValue(),
+            "The values for IN transform function must be single-valued");
+        _valueFunctions[i - 1] = valueFunction;
+      }
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 155 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.